### ISSUE TITLE
tsc-watch don't work on version 1.64 but work on 1.58.2

### ISSUE THREAD

**Issue by OrenVildermanPepperi** [microsoft/vscode#142626]
Issue Type: <b>Bug</b>

```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "Launch API Server",
            "skipFiles": [
                "<node_internals>/**"
            ],
            "cwd": "${workspaceFolder}/server-side",
            "program": "${workspaceFolder}/server-side/app.local.ts",
            // In the QA we prefer to debug on the Ts files, but this can be changed depending on what we are testing.
            // "outFiles": [
            //     "${workspaceFolder}/server-side/build/*/*.js"
            // ],
            "preLaunchTask": {
                "label": "Build API",
                "type": "typescript",
                "tsconfig": "server-side/tsconfig.json",
                "option": "watch",
                "problemMatcher": [
                    "$tsc-watch"
                ],
                "group": "build"
            },
            "runtimeExecutable": "${workspaceFolder}/server-side/node_modules/.bin/nodemon",
            "restart": true,
        },
        {
            "command": "cd ./server-side",
            "name": "Listen To NPM Scripts",
            "request": "launch",
            "type": "node-terminal"
        },
    ]
}
```

VS Code version: Code 1.64.1 (d6ee99e4c045a6716e5c653d7da8e9ae6f5a8b03, 2022-02-07T17:16:22.147Z)
OS version: Windows_NT x64 10.0.19044
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4770 CPU @ 3.40GHz (8 x 3399)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.68GB (4.43GB free)|
|Process Argv|--crash-reporter-id 38b09ac9-bb79-47da-872f-d55a4808f332|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (5)</summary>

Extension|Author (truncated)|Version
---|---|---
atlascode|atl|2.10.7
vscode-pull-request-github|Git|0.36.0
todo-tree|Gru|0.0.215
vscode-commons|red|0.0.6
vscode-yaml|red|1.4.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392cf:30425750
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30435082
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30433059
pythonvs932:30410667
wslgetstartedc:30433508
vscop940:30404999
vsclayoutctrc:30425945
vsrem710cf:30416617
9jga5758:30435146

```

</details>

**Comment by OrenVildermanPepperi** [comment]
Please notice that the crush happen when starting the: Launch API Server - With pressing F5,
This is the log I can see when debug start:
> Executing task in folder automation-api-tests: 'c:\Users\oren.v\Documents\new framwork\automation-api-tests\server-side\node_modules\.bin\tsc.cmd' -p 'c:\Users\oren.v\Documents\new framwork\automation-api-tests\server-side\tsconfig.json' --watch <

At line:1 char:101
+ ... utomation-api-tests\server-side\node_modules\.bin\tsc.cmd' -p 'c:\Use ...
+                                                                ~~
Unexpected token '-p' in expression or statement.
At line:1 char:104
+ ... tsc.cmd' -p 'c:\Users\oren.v\Documents\new framwork\automation-api-te ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Unexpected token ''c:\Users\oren.v\Documents\new framwork\automation-api-tests\server-side\tsconfig.json'' in expression or statement.
At line:1 char:194
+ ... \new framwork\automation-api-tests\server-side\tsconfig.json' --watch
+                                                                     ~~~~~
Unexpected token 'watch' in expression or statement.
At line:1 char:104
+ ... tsc.cmd' -p 'c:\Users\oren.v\Documents\new framwork\automation-api-te ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The '--' operator works only on variables or on properties.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : UnexpectedToken
 
The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command 'c:\Users\oren.v\Documents\new framwork\automation-api-tests\server-side\node_modules\.bin\tsc.cmd' -p 'c:\Users\oren.v\Documents\new framwork\automation-api-tests\server-side\tsconfig.json' --watch" terminated with exit code: 1.

**Comment by Corvax-Fisher** [comment]
I had a very similar issue with running C++ build tasks (which are just commands running in powershell).
The tasks fail to run in the new version 1.64.
I reinstalled 1.63.2 and there they work.

It looks like powershell commands in general don't work anymore.

In 1.63.2 I see that there is an ampersand and a space before the actual command that will be executed.
So this:
`Executing task in folder automation-api-tests: 'c:\Users\oren.v\Documents\new framwork\automation-api-tests\server-side\node_modules\.bin\tsc.cmd' -p 'c:\Users\oren.v\Documents\new framwork\automation-api-tests\server-side\tsconfig.json' --watch <`
Looks like this in 1.63.2:
`Executing task in folder automation-api-tests: & 'c:\Users\oren.v\Documents\new framwork\automation-api-tests\server-side\node_modules\.bin\tsc.cmd' -p 'c:\Users\oren.v\Documents\new framwork\automation-api-tests\server-side\tsconfig.json' --watch <`

I think this is needed to run CMD commands inside powershell, but I'm not 100 % sure on that.

**Comment by weinand** [comment]
@OrenVildermanPepperi please note, that VS Code does not support to inline prelaunchtasks in a launch config as you do above:

![CleanShot 2022-02-09 at 15 01 37@2x](https://user-images.githubusercontent.com/1898161/153220307-4f4140cf-b604-4c79-a772-ae8e3a30364c.png)

The fact that the inlined task is not flagged as an error is actually a bug in VS Code...

Please move the task into the tasks.json and use the task's name for the `preLaunchTask` property.

**Comment by weinand** [comment]
@OrenVildermanPepperi does your "Build API" task work fine after moving it to tasks.json?

**Comment by OrenVildermanPepperi** [comment]
@weinand I think its the same now, my prelaunchtasks config look like this:
`{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "Launch API Server",
            "skipFiles": [
                "<node_internals>/**"
            ],
            "cwd": "${workspaceFolder}/server-side",
            "program": "${workspaceFolder}/server-side/app.local.ts",
            // In the QA we prefer to debug on the Ts files, but this can be changed depending on what we are testing.
            // "outFiles": [
            //     "${workspaceFolder}/server-side/build/*/*.js"
            // ],
            "preLaunchTask": "Build API",
            "runtimeExecutable": "${workspaceFolder}/server-side/node_modules/.bin/nodemon",
            "restart": true,
        },
        {
            "command": "cd ./server-side",
            "name": "Listen To NPM Scripts",
            "request": "launch",
            "type": "node-terminal"
        },
    ]
}`
My tasks.json look like this:
`{
	"version": "2.0.0",
	"tasks": [
		{
			"label": "Build Addon",
			"type": "npm",
			"script": "build",
			"group": "build",
			"problemMatcher": []
		},
		{
			"label": "Build API",
			"type": "typescript",
			"tsconfig": "server-side/tsconfig.json",
			"option": "watch",
			"problemMatcher": [
				"$tsc-watch"
			],
			"group": "build"
		},
	]
}`

**Comment by weinand** [comment]
@OrenVildermanPepperi and if you now run the "Build API" task (outside of a launch configuration) via F1 > "Tasks: Run Task" does it work?

**Comment by OrenVildermanPepperi** [comment]
@weinand No but on any of these versions it will:
![image](https://user-images.githubusercontent.com/66742740/153243553-85829b38-2b76-4b95-949b-1827df1e2e05.png)
And on versions: 1.64.0 and 1.64.1 it will not

**Comment by weinand** [comment]
@OrenVildermanPepperi I see, thanks.
But the problem seems to be a task issue independent from debugging.

**Comment by alexr00** [comment]
Duplicate of https://github.com/microsoft/vscode/issues/142196, which will be fixed in 1.64.2.
